src/pyhf/constraints.py

```diff
@@ -1,25 +1,41 @@
 from . import get_backend, default_backend
 from . import events
+from .paramview import ParamViewer
 
 
 class gaussian_constraint_combined(object):
-    def __init__(self, pdfconfig):
+    def __init__(self, pdfconfig, batch_size=None):
+        self.batch_size = batch_size
         # iterate over all constraints order doesn't matter....
 
-        self.par_indices = list(range(len(pdfconfig.suggested_init())))
         self.data_indices = list(range(len(pdfconfig.auxdata)))
         self.parset_and_slice = [
             (pdfconfig.param_set(cname), pdfconfig.par_slice(cname))
             for cname in pdfconfig.auxdata_order
         ]
-        self._precompute()
-        events.subscribe('tensorlib_changed')(self._precompute)
 
-    def _precompute(self):
-        tensorlib, _ = get_backend()
+        pnames = [
+            cname
+            for cname in pdfconfig.auxdata_order
+            if pdfconfig.param_set(cname).pdf_type == 'normal'
+        ]
+
+        parfield_shape = (self.batch_size or 1, len(pdfconfig.suggested_init()))
+        self.parameter_helper = ParamViewer(parfield_shape, pdfconfig.par_map, pnames)
+
+        # self.parameter_helper.index_selection  is [ (batch, parslice) ] ]
```

Comment:
what's going on in this block of code?

Reply:
this sets up the `view` on the parameters needed by the constaint (when computing the constraint you do not need to access all parameters but just a subset of them, `ParamViewer` is a helper class that figures out the right indices to use